org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserProjectFileManager.java

```diff
@@ -1,15 +1,65 @@
 package org.code.javabuilder;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+
+/** Manages the in-memory list of user project files. */
 public class UserProjectFileManager implements ProjectFileManager {
+  private final String baseUrl;
+  private final String[] fileNames;
+  private final ArrayList<ProjectFile> fileList;
 
-  public UserProjectFileManager(String baseUrl, String[] fileNames) {}
+  public UserProjectFileManager(String baseUrl, String[] fileNames) {
+    this.baseUrl = baseUrl;
+    this.fileNames = fileNames;
+    this.fileList = new ArrayList<>();
+  }
 
+  /**
+   * Asynchronously loads all user project files.
+   *
+   * @throws UserFacingException If the load times out or fails.
+   */
   public void loadFiles() throws UserFacingException {
-    throw new UserFacingException("We couldn't load your files");
+    ExecutorService executor = Executors.newCachedThreadPool();
+    List<Callable<Boolean>> callables = new ArrayList<>();
+    for (String fileName : fileNames) {
+      ProjectFile file = new ProjectFile(fileName);
+      fileList.add(file);
+      callables.add(new ProjectFileLoader(file, String.join("/", baseUrl, fileName)));
+    }
+    List<Future<Boolean>> futures;
+    try {
+      futures = executor.invokeAll(callables);
+      executor.shutdown();
+      boolean finished;
+      finished = executor.awaitTermination(10, TimeUnit.SECONDS);
+
+      if (!finished) {
+        throw new UserFacingException(
+            "We couldn't fetch your files before our loader timed out. Try again.");
```

Comment:
(minor) I'm guessing we'll revisit all of these error strings at some point, but perhaps "the server" would be better than "our" or "our side"?

Reply:
Yeah, these are placeholders for now.